src/main/java/uk/gov/hmcts/reform/sendletter/config/ThreadPoolConfig.java

```diff
@@ -1,35 +1,126 @@
 package uk.gov.hmcts.reform.sendletter.config;
 
+import com.microsoft.applicationinsights.web.internal.RequestTelemetryContext;
+import com.microsoft.applicationinsights.web.internal.ThreadContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.Trigger;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
 import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import uk.gov.hmcts.reform.sendletter.util.TimeZones;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
 
 /**
  * Adds custom error handler to Scheduled Tasks.
  */
 @Configuration
-public class ThreadPoolConfig {
+@EnableScheduling
+public class ThreadPoolConfig implements SchedulingConfigurer {
 
-    private static int errorCount;
+    private static AtomicInteger errorCount = new AtomicInteger(0);
     private static final Logger log = LoggerFactory.getLogger(ThreadPoolConfig.class);
 
-    public static int getUnhandledTaskExceptionCount() {
-        return errorCount;
-    }
+    private static final Supplier<Long> CURRENT_MILLIS_SUPPLIER = () -> LocalDateTime
+        .now()
+        .atZone(ZoneId.of(TimeZones.EUROPE_LONDON))
+        .toInstant()
+        .toEpochMilli();
 
-    @Bean
-    public ThreadPoolTaskScheduler threadPoolTaskScheduler() {
-        ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+    private static final Supplier<RequestTelemetryContext> REQUEST_CONTEXT_SUPPLIER = () ->
+        new RequestTelemetryContext(CURRENT_MILLIS_SUPPLIER.get(), null);
 
-        threadPoolTaskScheduler.setThreadNamePrefix("SendLetterTask-");
-        threadPoolTaskScheduler.setErrorHandler(t -> {
+    @Override
+    public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+        ThreadPoolTaskScheduler taskScheduler = new SendLetterTaskScheduler();
+        taskScheduler.setThreadNamePrefix("SendLetterTask-");
+        taskScheduler.setErrorHandler(t -> {
             log.error("Unhandled exception during task. {}: {}", t.getClass(), t.getMessage(), t);
-            errorCount++;
+            errorCount.incrementAndGet();
         });
+        taskScheduler.initialize();
+
+        taskRegistrar.setTaskScheduler(taskScheduler);
+    }
+
+    /**
+     * Custom {@link ThreadPoolTaskScheduler} to be able to register scheduled tasks via AppInsights.
+     */
+    private static class SendLetterTaskScheduler extends ThreadPoolTaskScheduler {
```

Comment:
Worth putting a link in the comment https://github.com/Microsoft/ApplicationInsights-Java/wiki/Distributed-Tracing-in-Asynchronous-Java-Applications